build.sbt

```diff
@@ -1,110 +1,153 @@
-    Resolver.sonatypeRepo("snapshots")
-  ),
+/**
+ * Common settings
+ */
+lazy val commonSettings = sharedCommonSettings ++ Seq(
+  scalacOptions ++= commonScalacOptions,
+  parallelExecution in Test := false,
+  // resolvers += Resolver.sonatypeRepo("snapshots")
   libraryDependencies ++= Seq(
-    "com.github.mpilquist" %% "simulacrum" % "0.3.0",
-    "org.typelevel" %% "machinist" % "0.3.0",
-    compilerPlugin("org.scalamacros" % "paradise" % "2.1.0-M5" cross CrossVersion.full),
-    compilerPlugin("org.spire-math" %% "kind-projector" % "0.6.3")
-  ),
-  scmInfo := Some(ScmInfo(url("https://github.com/non/alleycats"),
-    "scm:git:git@github.com:non/alleycats.git")),
-  commands += gitSnapshots
-)
+    "com.github.mpilquist" %%% "simulacrum" % simulacrumVersion,
+    "org.typelevel" %%% "machinist" % machinistVersion,
+    compilerPlugin("org.spire-math" %% "kind-projector" % kindProjectorVersion)
+  )
+) ++ warnUnusedImport ++ unidocCommonSettings
 
-lazy val alleycatsSettings = buildSettings ++ commonSettings ++ publishSettings ++ scoverageSettings
+lazy val commonJsSettings = Seq(
+  scalaJSStage in Global := FastOptStage
+)
 
-lazy val disciplineDependencies = Seq(
-  "org.scalacheck" %% "scalacheck" % "1.12.4",
-  "org.typelevel" %% "discipline" % "0.3"
+lazy val commonJvmSettings = Seq(
+ // testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-oDF)
 )
 
+/**
+ * Alleycats - This is the root project that aggregates the alleycatsJVM and alleycatsJS sub projects
+ */
+lazy val alleycatsSettings = buildSettings ++ commonSettings ++ publishSettings ++ scoverageSettings
+
 lazy val alleycats = project.in(file("."))
+  .settings(moduleName := "root")
+  .settings(alleycatsSettings)
+  .settings(noPublishSettings)
+  .settings(console <<= console in (alleycatsJVM, Compile))
+  .aggregate(alleycatsJVM, alleycatsJS)
+  .dependsOn(alleycatsJVM, alleycatsJS, testsJVM % "test-internal -> test")
+
+lazy val alleycatsJVM = project.in(file(".alleycatsJVM"))
   .settings(moduleName := "alleycats")
   .settings(alleycatsSettings)
-  .aggregate(core, laws, tests)
-  .dependsOn(core, laws, tests % "test-internal -> test")
+  .settings(commonJvmSettings)
+  .aggregate(coreJVM, lawsJVM, testsJVM)
+  .dependsOn(coreJVM, lawsJVM, testsJVM % "test-internal -> test")
 
-lazy val core = project
-  .settings(moduleName := "alleycats-core")
+lazy val alleycatsJS = project.in(file(".alleycatsJS"))
+  .settings(moduleName := "alleycats")
   .settings(alleycatsSettings)
-  .settings(libraryDependencies += "org.spire-math" %% "cats-core" % "0.2.0")
+  .settings(commonJsSettings)
+  .aggregate(coreJS, lawsJS, testsJS)
+  .dependsOn(coreJS, lawsJS, testsJS % "test-internal -> test")
+  .enablePlugins(ScalaJSPlugin)
+
+/**
+ * Core project
+ */
+lazy val core = crossProject.crossType(CrossType.Pure)
+  .settings(moduleName := "alleycats-core")
+  .settings(alleycatsSettings:_*)
+  .settings(scalaMacrosParadise(paradiseVersion):_*)
```

Comment:
ah then all 3 of them could do with the change, aka https://github.com/sbt/sbt/pull/1902/files